test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/compiler/resources/uwpBlankAppV2.json

```diff
@@ -104,12 +106,18 @@
           "ref/netcore50/System.Numerics.dll": {},
           "ref/netcore50/System.Runtime.Serialization.dll": {},
           "ref/netcore50/System.ServiceModel.Web.dll": {},
-          "ref/netcore50/System.ServiceModel.dll": {},
+          "ref/netcore50/System.ServiceModel.dll": {
+            "related": ".Web.dll"
+          },
           "ref/netcore50/System.Windows.dll": {},
           "ref/netcore50/System.Xml.Linq.dll": {},
           "ref/netcore50/System.Xml.Serialization.dll": {},
-          "ref/netcore50/System.Xml.dll": {},
-          "ref/netcore50/System.dll": {},
+          "ref/netcore50/System.Xml.dll": {
+            "related": ".Linq.dll;.Serialization.dll"
```

Comment:
I think should be a JSON array instead of a semi-colon delimited list

Reply:
Thanks! This is a good question!
This related property will be added in a [LockFileItem](https://github.com/NuGet/NuGet.Client/blob/4fef99532f4022504feec5f68c8501cbeadd3aed/src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileItem.cs) and a LockFileItem is as below:
```
public string Path { get; }
public IDictionary<string, string> Properties { get; } = new Dictionary<string, string>();
```
So if we want to keep the LockFileItem unchanged, the content of related property has to be a string I think.
Or else, it might be a breaking change.
The spec mentioned this at [here](https://github.com/NuGet/Home/blob/dev/proposed/2021/NuGetConsumePDBInPackageReference.md#project-reference).

Reply:
From a technical point of view, `LockFileItem` could use a `JsonConverter` to put `related` into `IReadOnlyList<string> Related { get; }`, but all other properties into the `Properties` property. It means that the `related` property value isn't in the `Properties` dictionary, and that everything reading the assets file will also need to special case this property. But realistically, consumers are going to need have a special code path just for `related` anyway, since the way they treat that property is different to how they will treat other properties anyway.

Reply:
I tested as following. If we change it into array, the old build task will have error in loading the new lockfile.
1.Create a classlib project, install newtonsoft.json package.
2.Changed the project.assets.json file from:
```
"targets": {
    "net5.0": {
      "Newtonsoft.Json/13.0.1": {
        "type": "package",
        "compile": {
          "lib/netstandard2.0/Newtonsoft.Json.dll": {}
        },
        "runtime": {
          "lib/netstandard2.0/Newtonsoft.Json.dll": {}
        }
      }
    }
  },
```
to:
```
 "targets": {
    "net5.0": {
      "Newtonsoft.Json/13.0.1": {
        "type": "package",
        "compile": {
          "lib/netstandard2.0/Newtonsoft.Json.dll": {
            "related": [".pdb", ".xml"]
          }
        },
        "runtime": {
          "lib/netstandard2.0/Newtonsoft.Json.dll": {
            "related": [".pdb", ".xml"]
          }
        }
      }
    }
  },
```
3.Run dotnet build on this test project, failed for:
```
C:\Program Files\dotnet\sdk\7.0.100-preview.4.22252.9\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(267,5): error NETSDK1060: Error reading assets file: Error loading lock file 'C:\Users\henli\Source\Repos
\TestPDB\TestPDB\obj\project.assets.json' : Cannot cast Newtonsoft.Json.Linq.JArray to Newtonsoft.Json.Linq.JToken. [C:\Users\henli\Source\Repos\TestPDB\TestPDB\TestPDB.csproj]
```

Reply:
Oops, yes, sorry. I forgot about the problems with mismatched versions. It is indeed very difficult for us to make breaking changes (improvements). 😞